app/controllers/admin/export/document_controller.rb

```diff
@@ -24,6 +27,24 @@ def edition_associations(edition)
     associations.each do |association|
       output[:associations][association] = edition.public_send(association)
     end
+    output[:whitehall_admin_links] = resolve_whitehall_admin_links(edition.body)
     output
   end
+
+  def resolve_whitehall_admin_links(body)
+    whitehall_links = whitehall_admin_links(body)
+    public_urls = []
+    whitehall_links.each do |link|
+      public_url = public_url_for_admin_link(link)
+      public_urls.push("whitehall_admin_url": link, "public_url": public_url)
+    end
```

Comment:
This looks like it'd more idiomatic as a map rather than an each iteration. 

Could get the whole method down to `whitehall_admin_links(body).map { |link| { whitehall_admin_url: link, public_url: public_url_for_admin_link(link) }`